test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -109,13 +110,96 @@ class cache_limit_statistic : public statistic {
     int64_t limit;
 };
 
+static std::string
+collection_name_to_file_name(const std::string &collection_name)
+{
+    /* Strip out the URI prefix. */
+    const size_t colon_pos = collection_name.find(':');
+    const auto stripped_name =
+      (colon_pos == std::string::npos) ? collection_name : collection_name.substr(colon_pos + 1);
+
+    /* Now add the directory and file extension. */
+    return std::string(DEFAULT_DIR) + "/" + stripped_name + ".wt";
+}
+
+class db_size_statistic : public statistic {
+    public:
+    db_size_statistic(configuration *config, database &database)
+        : statistic(config), _database(database)
+    {
+        _limit = config->get_int(LIMIT);
+#ifdef _WIN32
+        debug_print("Database size checking is not implemented on Windows", DEBUG_ERROR);
+#endif
+    }
+    virtual ~db_size_statistic() = default;
+
+    /* Don't need the stat cursor for this. */
+    void
+    check(WT_CURSOR *) override final
+    {
+        const auto file_names = get_file_names();
+#ifndef _WIN32
+        size_t db_size = 0;
+        for (const auto &name : file_names) {
+            struct stat sb;
+            if (stat(name.c_str(), &sb) == 0) {
+                db_size += sb.st_size;
+                debug_print(name + " was " + std::to_string(sb.st_size) + " bytes", DEBUG_TRACE);
+            } else
+                /* The only good reason for this to fail is if the file hasn't been created yet. */
+                testutil_assert(errno == ENOENT);
+        }
+        debug_print("Current database size is " + std::to_string(db_size) + " bytes", DEBUG_TRACE);
+        if (db_size > _limit) {
+            const std::string error_string =
+              "runtime_monitor: Database size limit exceeded during test! Limit: " +
+              std::to_string(_limit) + " db size: " + std::to_string(db_size);
+            debug_print(error_string, DEBUG_ERROR);
```

Comment:
Should we use tesutil_die instead ?

Reply:
I would like to discuss a better error handling methodology in the test framework in the coming week but for now I think this is fine.

Reply:
@tetsuo-cpp please replace with testutil_die(-1, error_string.c_str()).

Reply:
b1728fe